File: player-association-client/src/types/Insight.ts
export type InsightCategory =
  | "Development"
  | "Health & Fitness"
  | "Career & Education"
  | "Transfer Market"
  | "Interview"
  | "Biography";

export interface Insight {
  id: number;
  name: string;
  category: InsightCategory;
  description: string;
  content: string;
  imagePath: string;
  createdAt: string;
  updatedAt?: string;
}

File: PlayerAssociationAPI/Models/Insight.cs
using System;

namespace PlayerAssociationAPI.Models
{
    public class Insight
    {
        public int Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string ImagePath { get; set; } = string.Empty;

        public string Content { get; set; } = string.Empty;

        public string Author { get; set; } = string.Empty;

        public InsightCategory Category { get; set; } // Store as enum in DB

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}